plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/exporter/DebugExporter.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.core.exporter;
+
+import org.opensearch.plugin.insights.rules.model.SearchQueryRecord;
+
+import java.util.List;
+
+/**
+ * Debug exporter for development purpose
+ */
+public final class DebugExporter extends AbstractExporter {
+    /**
+     * Constructor of DebugExporter
+     */
+    public DebugExporter() {}
+
+    /**
+     * Write the list of SearchQueryRecord to debug log
+     *
+     * @param records list of {@link SearchQueryRecord}
+     * @return true
+     */
+    @Override
+    public boolean export(final List<SearchQueryRecord> records) {
+        logger.debug("QUERY_INSIGHTS_RECORDS: " + records.toString());
+        return true;
+    }
```

Comment:
This probably allocates significant memory, just for logging. Also, should we check if debug logging is enabled or not?

Reply:
Lazily building suppliers is also good option - https://logging.apache.org/log4j/2.x/javadoc/log4j-api/org/apache/logging/log4j/Logger.html#debug(java.lang.String,org.apache.logging.log4j.util.Supplier...)

Reply:
The debug exporter will only be enabled for debuggig and testing, it won't be enabled by default. In this case do we still need to worry about memory usage?

Reply:
Should be fine in that case. Although, I hope we don't enable it for debugging functional and not resiliency issues